src/Data/StateVar.hs

```diff
@@ -7,6 +7,12 @@
 {-# LANGUAGE DefaultSignatures #-}
 #endif
 {-# LANGUAGE TypeFamilies #-}
+-- Foreign.ForeignPtr is unsafe before GHC-7.10
+#if __GLASGOW_HASKELL__ >=710
```

Comment:
If this depends on the API in `Foreign.ForeignPtr`, shouldn't this use `MIN_VERSION_base` rather than `__GLASGOW_HASKELL__`?

Reply:
then it should be `__GLASGOW_HASKELL__ >= 704 && MIN_VERSION_base(4,8,0)`. I can change it to be that.

Reply:
Looks OK to me. Just for fun, I tried to edit this directly in the GitHub PR, let's see how this works...

Reply:
Editing online seems to have worked, I can make a new release if the current state is OK for you...